rllib/execution/parallel_requests.py

```diff
@@ -182,3 +183,224 @@ def wait_asynchronous_requests(
+            AsyncRequestsManager.get_ready_requests().
+
+    Example:
+        >>> import time
+        >>> import ray
+        >>> from ray.rllib.execution.parallel_requests_manager import (
+        ...     AsyncRequestsManager)
+        >>>
+        >>> @ray.remote
+        ... class MyActor:
+        ...    def apply(self, fn, *args: List[Any], **kwargs: Dict[str, Any]) -> Any:
+        ...        return fn(*args, **kwargs)
+        ...
+        ...    def task(self, a: int, b: int) -> Any:
+        ...        time.sleep(0.5)
+        ...        return a + b
+        >>>
+        >>> workers = [MyActor.remote() for _ in range(3)]
+        >>> manager = AsyncRequestsManager(workers,
+        ...                                max_remote_requests_in_flight_per_actor=2)
+        >>> manager.submit(lambda worker, a, b: worker.task(a, b), fn_args=[1, 2])
+        >>> print(manager.get_ready_requests())
+        >>> manager.submit(lambda worker, a, b: worker.task(a, b),
+        ...                fn_kwargs={"a": 1, "b": 2})
+        >>> time.sleep(2) # Wait for the tasks to finish.
+        >>> print(manager.get_ready_requests())
+    """
+
+    def __init__(
+        self,
+        workers: List[ActorHandle],
+        max_remote_requests_in_flight: int = 2,
+        ray_wait_timeout_s: Optional[float] = 0.03,
+        return_object_refs: bool = False,
+    ):
+        self._ray_wait_timeout_s = ray_wait_timeout_s
+        self._return_object_refs = return_object_refs
+        self._max_remote_requests_in_flight = max_remote_requests_in_flight
+        self._all_workers = set(workers)
+        self._num_workers = len(workers)
+        self._available_workers = workers.copy()
+        self._available_workers_set = set(self._available_workers)
+        self._unavailable_workers = set()
+        self._remote_requests_in_flight = defaultdict(set)
+        self._pending_to_actor = {}
+        self._call_queue = Queue(maxsize=0)
+        self._pending_remotes = []
+
+    def submit(
+        self,
+        remote_fn: Callable,
+        *,
+        fn_args: List[Any] = None,
+        fn_kwargs: Dict[str, Any] = None,
+        for_all_workers: bool = False,
+    ) -> None:
+        """Submit a remote function call schedule on available workers.
+
+        Args:
+            remote_fn: The remote function to call
+            fn_args: The arguments to pass to the remote function
+            fn_kwargs: The keyword arguments to pass to the remote function
+            for_all_workers: If True, submit this request to the queue for as many
+                number of workers that there are. If False, submit the request without
+                replicating it.
+
+        """
+        if fn_args is None:
+            fn_args = []
+        if fn_kwargs is None:
+            fn_kwargs = {}
+        call = (remote_fn, fn_args, fn_kwargs)
+        if for_all_workers:
+            for _ in range(self._num_workers):
+                self._call_queue.put(call)
```

Comment:
how do you guarantee these N calls get dispatched to all the workers? instead of having some actors handle the call multiple times?